mqtt/mqtt-broker/benches/dispatch_messages.rs

```diff
@@ -0,0 +1,331 @@
+                                );
+                            }
+                            None
+                        }
+                        ClientEvent::PubRec(_) => None,
+                        ClientEvent::PubRel(_) => None,
+                        ClientEvent::PubComp(_) => None,
+                        ClientEvent::DropConnection
+                        | ClientEvent::CloseSession
+                        | ClientEvent::Disconnect(_) => {
+                            stop = true;
+                            None
+                        }
+                        ClientEvent::PingReq(_) => Some(ClientEvent::PingResp(proto::PingResp)),
+                        ClientEvent::PingResp(_) => Some(ClientEvent::PingReq(proto::PingReq)),
+                        ClientEvent::ConnReq(_) => None,
+                        ClientEvent::ConnAck(_) => None,
+                        ClientEvent::Subscribe(_) => None,
+                        ClientEvent::SubAck(_) => None,
+                        ClientEvent::Unsubscribe(_) => None,
+                        ClientEvent::UnsubAck(_) => None,
+                    };
+
+                    if let Some(event) = event_out {
+                        let message = Message::Client(client_id.clone(), event);
+                        if let Err(e) = self.broker_handle.send(message).await {
+                            warn!("{}: Broker closed connection. {:?}", client_id, e);
+                            stop = true;
+                        }
+                    }
+                }
+                Message::System(_) => {}
+            };
+
+            if stop {
+                info!("{}: disconnect requested", self.client_id);
+                break;
+            }
+        }
+
+        info!(
+            "{}: received {} messages",
+            self.client_id, messages_received
+        );
+    }
+}
+
+struct PublishHandle {
+    client_id: ClientId,
+    pubs_to_be_acked: Arc<Mutex<HashSet<proto::PacketIdentifier>>>,
+}
+
+impl PublishHandle {
+    fn publish(&self, qos: proto::QoS, topic_name: String, payload_size: Size) -> proto::Publish {
+        let packet_id = match qos {
+            proto::QoS::AtMostOnce => proto::PacketIdentifierDupQoS::AtMostOnce,
+            proto::QoS::AtLeastOnce => {
+                let mut acks = self.pubs_to_be_acked.lock().expect("pubs_to_be_acked");
+
+                let mut next = proto::PacketIdentifier::new(1).unwrap();
+                while !acks.insert(next) {
+                    next += 1
+                }
+
+                proto::PacketIdentifierDupQoS::AtLeastOnce(next, false)
+            }
+            proto::QoS::ExactlyOnce => todo!(),
+        };
+
+        proto::Publish {
+            packet_identifier_dup_qos: packet_id,
+            retain: false,
+            topic_name,
+            payload: Bytes::from_iter((0..payload_size.into()).map(|_| rand::random::<u8>())),
```

Comment:
nit: I'm not sure if randomness does matter here. Won't `vec![0u8, size]` spare some cycles?

Reply:
1. data preparation doesn't participate in execution measurements., only message publishing 
2. randomization help to make cache missed when loading that "emulates" real behavior.